pkg/agent/controller/egress/egress_controller.go

```diff
@@ -671,14 +713,18 @@ func (c *EgressController) syncEgress(egressName string) error {
 		}
 	}
 
+	// rateLimitChanged will be true if this Egress update its rateLimit or this is a new Egress.
+	rateLimitChanged := eState.hasRateLimit != (egress.Spec.RateLimit != nil)
```

Comment:
how is this going to evaluate to true when Egress rate limit is updated?

Reply:
Not exactly. It is true if:
1. a new Egress with rate-limit is created
2. an existing Egress removes its rate-limit
3. an existing Egress adds rate-limit from non-rate-limit

Let me change the var name and update the comments to reduce the confusion.